### ISSUE TITLE
Kernel dies during training [memory issue]

### ISSUE THREAD

**Issue by vash8120** [DeepLabCut/DeepLabCut#375]
Windows 10 Enterprise (CPU), Python 3 (Latest Anaconda version), Deeplabcut 2.0
OS: [Microsoft Windows 10 Enterprise; Version 10.0.17134 Build 17134]
DeepLabCut Version: ['2.0.7.2'] 
Browser [Microsoft Edge]

Problem: During training, kernel dies after around 900 iterations. We are running it on WindowsCPU, and saving every 300 iterations

1. Follow the instructions to easy-install for WindowsCPU environment
2. Create project, label frames and create training dataset
3. Run [deeplabcut.train_network(path_config_file,displayiters=100,saveiters=300)]
4. After around 900 iterations: [Error: Kernel restarting. The kernel appears to have died. It will restart automatically]

5 Kernel does not seem to restart, and saved variables and imported packages are lost. Needs to be manually restarted and packages reloaded.

**Screenshots**
<img width="892" alt="kernelDead2" src="https://user-images.githubusercontent.com/53270701/61810672-b706e100-ae0d-11e9-9791-965302c46689.PNG">

**Additional context**
Since it saves every 300 iterations, we can manually ask it to train using the most recent iteration (by changing the pose config file init weights from resnet to current iteration). This is extremely time-consuming since it will fail every few 100 iterations.

**Comment by MMathisLab** [comment]
unfortunately this seems to be a Jupyter related issue; on terminal we do not see this error.* unless a memory full issue ;)

**Comment by vash8120** [comment]
The same issue also comes up in Spyder
Shall I try running it from Anaconda PowerShell directly? Would the commands remain the same here as for Jupyter notebook?

**Comment by MMathisLab** [comment]
Yes, all commands are the same :) Check out the docs, [here](https://github.com/AlexEMG/DeepLabCut/blob/master/docs/functionDetails.md), and the user-guide which walks you through the 12 steps: https://www.nature.com/articles/s41596-019-0176-0 

not sure why the kernel would be dying, unless you are using all the memory; perhaps check your system monitor when you are running the code....

**Comment by vash8120** [comment]
Same issue on Terminal.
It might be a memory problem. Total memory is 32Gb. It counts down in memory per iteration, until the kernel dies (See before and after crash pictures below)

<img width="1179" alt="KernelMemory" src="https://user-images.githubusercontent.com/53270701/61818586-2507d400-ae1f-11e9-9403-9bf0e268c8ed.PNG">
<img width="1508" alt="KernelMemoryAfter" src="https://user-images.githubusercontent.com/53270701/61818595-29cc8800-ae1f-11e9-95f9-168459b52a70.PNG">

Does this mean the system cannot handle the size of the training, or can we edit the code to overwrite old models to free up memory? How much memory would we need for running it on WindowsCPU? What would you recommend?
This video is 3.4 Gb in size (520 width by 450 height), 5200 frames long. We used 200 labeled frames.

**Comment by MMathisLab** [comment]
The training code does not use *any* video. The network scales with the size of the image; please see discussion here: https://www.nature.com/articles/s41593-018-0209-y and here: https://www.nature.com/articles/s41596-019-0176-0 

In short, you might want to make your images smaller. We stress that you should crop your images to remove any unnecessary pixels, see CRITICAL POINT: https://github.com/AlexEMG/DeepLabCut/blob/master/docs/functionDetails.md#c-data-selection

**Comment by MMathisLab** [comment]
You would be better off training on Colab then; you just move your whole project folder to google drive, edit only the config.yaml ``project_path`` file, then use this: https://github.com/AlexEMG/DeepLabCut/blob/master/examples/Colab_TrainNetwork_VideoAnalysis.ipynb

**Comment by AlexEMG** [comment]
Are you perhaps running other jobs in TF or other programs. What is your TF version. (the frame size you describe should not be a problem at all)

**Comment by vash8120** [comment]
@AlexEMG TF version is 1.12.0 (it crashed within 1100 iterations this time)
@MMathisLab I'll try cropping to see if it helps. I'm trying Colab right now. Will keep you updated. Thanks!

**Comment by MMathisLab** [comment]
HI @vash8120 any updates?

**Comment by vash8120** [comment]
Colab is working well! 
I haven't tried cropping the images to run it on windowsCPU yet. I'm not sure if the memory problem is a Tensorflow issue or a system memory capacity problem. We may just use Colab going forward.
Thanks!
